### ISSUE TITLE
Multiple PID providers and "legacy" provider since v6.2: extend documentation, change code for :DoiProvider

### ISSUE THREAD

**Issue by vkush** [IQSS/dataverse#10658]
**Overview of the Feature Request**
Since Dataverse v6.2 usage of multiple PID providers is supported. Documentation also allows to use [backward-compatibility for a single PID provider](https://guides.dataverse.org/en/6.2/installation/config.html#backward-compatibility-for-single-pid-provider-installations) using specific of each PID type JVM options and database settings (documentation is using wording "Legacy Single PID provider" in front of the options). Dataverse v6.2 has also new JVM options "dataverse.pid.providers" and "dataverse.pid.default-provider" which must be always configured (if they are empty - exception occurs in log). The problem is, that documentation does not mention the fact, that even when a legacy approach with a single PID provider is used these 2 options must be set. From the [discussion in user group](https://groups.google.com/g/dataverse-community/c/ZyhO54in3Jo) comes info, that special PID provider "legacy" can be used in the configuration ([see this reply-1](https://groups.google.com/g/dataverse-community/c/ZyhO54in3Jo/m/hda3AJmuEAAJ)), what was tested and works well ([see this reply-2](https://groups.google.com/g/dataverse-community/c/ZyhO54in3Jo/m/U1b5X9oICAAJ)). 

Based on this the issue consist of 2 points:

1) Extend documentation for backward compatibility explaining possibility of usage of a "legacy" string in options "dataverse.pid.providers" and "dataverse.pid.default-provider" e.g. like in the reply-1 above.

2) Slightly adapt code for analysis of ":DoiProvider" database setting because of problem in reply-2 above, that ":DoiProvider" currently must be non-null when legacy approach is used, even for non-DOI PID providers. See [code here with next "if"](https://github.com/IQSS/dataverse/blob/3dccdb7506bf642a4d331ae00ca3f52cbbd4baf1/src/main/java/edu/harvard/iq/dataverse/pidproviders/PidProviderFactoryBean.java#L147) and also [reply-3 here](https://groups.google.com/g/dataverse-community/c/ZyhO54in3Jo/m/PXrOGw0MCAAJ).

Since support of multiple PID providers in v6.2 is working correct this issue can be considered as not critical and only for the time, when backward-compatibility for a single PID provider is supported.

**What kind of user is the feature intended for?**
all

**What inspired the request?**
Extension of the documentation

**What existing behavior do you want changed?**

**Any brand new behavior do you want to add to Dataverse?**

**Any open or closed issues related to this feature request?**
See discussion in the user group linked above.

**Comment by pdurbin** [comment]
Yes. We added default-provider in the code at d441361fbca99190260eca70a9aa082f5d9d2a80 (from #10563), for example, but we should clarify the docs!

I haven't fully groked the :DoiProvider issue. 😅 

@vkush any interest in making a pull request? 😄

**Comment by sbarbosadataverse** [comment]
@vkush Hello. Are you interested in making a pull request for this feature enhancement? We are reviewing feature issues we can keep active, so your feedback is appreciated. Thanks

**Comment by vkush** [comment]
I'm sorry, but I can't create a proper pull request because I do not have an instance for development, to check that code works.

Issue for me is not relevant, because I am using the new way of the PID description. This issue could be relevant ONLY for instances, where the old way of PID description is still used - it should be mentioned in the documentation that "legacy" PID provider can be set/described via JVM options or can be chosen then via GUI (see my point 1 above); and also ":DoiProvider" must be ALWAYS not-null (what will be set via documentation for DOIs (https://guides.dataverse.org/en/6.3/installation/config.html#configuring-your-dataverse-installation-for-a-single-doi-provider, see "Database settings"), but it will not be set (at least explicitly) for other types of PID, e.g. Handles (see my point 2 above, especially "reply-3", where some issue with ":DoiProvider" is referenced for old versions 5.x).

For the issue with ":DoiProvider" some code can be changed (https://github.com/IQSS/dataverse/blob/3dccdb7506bf642a4d331ae00ca3f52cbbd4baf1/src/main/java/edu/harvard/iq/dataverse/pidproviders/PidProviderFactoryBean.java#L149):

from

```
        if (protocol != null && authority != null && shoulder != null && provider != null) {
            logger.warning("Found legacy settings: " + protocol + " " + authority + " " + shoulder + " " + provider
                    + "Please consider updating as this PIDProvider configuration mechanism will be removed in a future version of Dataverse");
```

to

```
        if (protocol != null && authority != null && shoulder != null && protocol != "doi") || 
           (protocol != null && authority != null && shoulder != null && provider != null && protocol == "doi") {
            if (provider != null) {
	            logger.warning("Found legacy settings: " + protocol + " " + authority + " " + shoulder + " " + provider + ". "
        	            + "Please consider updating as this PIDProvider configuration mechanism will be removed in a future version of Dataverse");
            } else {
		    logger.warning("Found legacy settings: " + protocol + " " + authority + " " + shoulder + ". "
        	            + "Please consider updating as this PIDProvider configuration mechanism will be removed in a future version of Dataverse");
            }
```

I have added:
1) That ":DoiProvider" (String "provider") should be checked only for "doi" protocol.
2) ". " was added for better warning output (not sure if it's needed).

**Comment by cmbz** [comment]
2025-07-14
- Reviewed, keeping
- Documentation is desired. Other elements maybe valuable, too.
